src/multicorn.c

```diff
@@ -1453,6 +1560,391 @@ multicornIsForeignScanParallelSafe(PlannerInfo *root, RelOptInfo *rel,
 }
 #endif
 
+/*
```

Comment:
That's a lot of code! Can you mark the parts taken from other FDWs (here and in deparse) and parts that you added yourself so that I know where to concentrate the review? Currently it kind of makes sense to me but knowing where it came from would make it clearer.

Reply:
Sure, I can do that. I can add some comments like `// MY CODE START` and `// MY CODE END` if that helps. Just keep in mind that the parts taken from other FDWs are also trimmed down, i.e. I've thrown away the irrelevant stuff so it's not 1-1.

Reply:
Done - I've enclosed all deviations from common FDW code (as used in `postgres_fdw` and other implementations) with the above comments in `multicorn.c` and `deparse.c` (other files should be more easier to parse I think).

Again worth mentioning that common FDW code that I've "appropriated" was pruned.